### PR TITLE
Change mutable arrays to immutable lists in Queries

### LINKED ISSUE #114: Change final arrays to unmodifiable lists
For immutability of queries for example.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Closes #114 

@nikitin-da please review

### CODE REVIEW

File: storio/src/test/java/com/pushtorefresh/storio/util/QueryUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            assertEquals(list.get(i), unmodifiableList.get(i));
+        }
+    }
+
+    @SuppressWarnings("ConstantConditions") @Test public void listToUnmodifiableIsReallyUnmodifiable() {
+        List<String> unmodifiableList = QueryUtil.listToUnmodifiable(Arrays.asList("1", "2", "3"));
+
+        // UnmodifiableCollection is private class :(
+        String className = unmodifiableList.getClass().getSimpleName();
+        assertTrue(className.equals("UnmodifiableRandomAccessList") || className.equals("UnmodifiableList"));
```
Comment (nikitin-da) [inline storio/src/test/java/com/pushtorefresh/storio/util/QueryUtilTest.java]: hehe

File: storio/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import android.support.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Util methods for queries
+ */
+public class QueryUtil {
```
Comment (nikitin-da) [inline storio/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java]: May it be final and forbid extension?

**Comment by nikitin-da** [comment]
Well done!

File: storio/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java (hunk 2 above)
Comment (artem-zinnatullin) [inline storio/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java]: It's not necessary, I made constructor `private` :)

**Pull request merged**